### PR TITLE
Collect bundled plugin and locally added gem

### PR DESCRIPTION
**Pull request by kenhys**
It is useful to detect additionally installed plugin.

Case 1: planning to migrate from td-agent to fluent-package. manually installed plugin must reinstall for fluent-package. so collect such information by fluent-diagtool.

 [Diagtool] [INFO] [Collect] Collecting fluent-package gem information...
 [Diagtool] [INFO] [Collect] fluent-package gem information is stored in tmp/20230901075350/output/tdgem_list.output
 [Diagtool] [INFO] [Collect] fluent-package gem information (bundled by default) is stored in tmp/20230901075350/output/gem_bundled_list.output
 [Diagtool] [INFO] [Collect] fluent-package manually installed gem information is stored in tmp/20230901075350/output/gem_local_list.output
 [Diagtool] [INFO] [Collect] fluent-package manually installed gems:
 [Diagtool] [INFO] [Collect]   * fluent-plugin-concat

### CODE REVIEW

**Review by daipom — APPROVED** [review]
Thanks! LGTM.
I confirmed this can collect the plugin-list manually installed.

**Comment by kenhys** [comment]
@kubotat 

We are planning to release fluent-package LTS v5.0.2 in Nov, 29 and want to ship new version of diagtool which contains #14, #15, #16.

Can we (@kenhys  or @ashie ) merge and release gem? 
If you willing to review above PR, we'll wait for your feedback, if not, we will go ahead.
(We have already merge rights and ownership of gem, but we want to respect your intention)

**Comment by kubotat** [comment]
@kenhys 
> Can we (@kenhys or @ashie ) merge and release gem?

Yes, please merge it and release new gem. Thanks for adding new feature.

**Comment by kenhys** [comment]
@kubotat Thank you for quick answer!

**Pull request merged**
